### PR TITLE
Implement several NR system tests based on examples

### PR DESCRIPTION
**Pull request by DavidCurtiss**
### What does this Pull Request accomplish?

Adds system level tests based on some of the NR C Examples.

Related work item: [AB#1794282](https://ni.visualstudio.com/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/1794282).

### Why should this Pull Request be merged?

Spot check the NR service.

### What testing has been done?

image

### CODE REVIEW

File: source/tests/system/nirfmxnr_driver_api_tests.cpp [hunk 1]
```diff
@@ -68,12 +61,55 @@ class NiRFmxNRDriverApiTests : public Test {
   std::unique_ptr<NiRFmxNR::Stub> stub_;
 };
 
+#define CHECK_ERROR(session) if (1) \
```
Comment (astarche) [inline source/tests/system/nirfmxnr_driver_api_tests.cpp:64]: +1 for implementing CHECK/EXPECT methods these as macros. I haven't been doing that, but any time you can put an assert wrapper in a macro, it preserves the line number of the caller. That helps a ton with debugging failures.

**Review by astarche — APPROVED** [review]

File: source/tests/system/nirfmxnr_driver_api_tests.cpp (hunk 1 above)
Comment (reckenro) [reply source/tests/system/nirfmxnr_driver_api_tests.cpp:64]: Another +1 for the macros! I was wondering what the advantage of them were at first but that's really handy.

File: source/tests/system/nirfmxnr_driver_api_tests.cpp [hunk 2]
```diff
@@ -87,21 +123,622 @@ TEST_F(NiRFmxNRDriverApiTests, Init_Close_Succeeds)
...
+  EXPECT_TRUE(isnan(PDSCHRMSEVMMean[1]));
+  EXPECT_EQ(0.0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[0].x0());
+  EXPECT_EQ(0.0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[0].dx());
+  EXPECT_EQ(0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[0].rms_evm_per_subcarrier_mean_size());
+  EXPECT_EQ(0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[0].rms_evm_per_subcarrier_mean().size());
+  EXPECT_EQ(0.0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[1].x0());
+  EXPECT_EQ(0.0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[1].dx());
+  EXPECT_EQ(0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[1].rms_evm_per_subcarrier_mean_size());
+  EXPECT_EQ(0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response[1].rms_evm_per_subcarrier_mean().size());
+  // EXPECT_EQ(0.0, mod_acc_fetch_rmsevm_per_subcarrier_mean_trace_response.rms_evm_per_subcarrier_mean(0));
```
Comment (reckenro) [inline source/tests/system/nirfmxnr_driver_api_tests.cpp:443]: Should we just delete the commented out asserts on the first element when the array is empty?

**Review by reckenro — APPROVED** [review]

File: source/tests/system/nirfmxnr_driver_api_tests.cpp (hunk 2 above)
Comment (DavidCurtiss) [reply source/tests/system/nirfmxnr_driver_api_tests.cpp:443]: Similar to your new WLAN tests, I'm holding out hope that we can figure out the causes for the errors were getting, and fix them.

File: source/tests/system/nirfmxnr_driver_api_tests.cpp [hunk 3]
```diff
@@ -87,21 +123,622 @@ TEST_F(NiRFmxNRDriverApiTests, Init_Close_Succeeds)
...
+
+  EXPECT_EQ(0.0, auto_level_response.reference_level());
+  EXPECT_LT(0.0, fetch_subblock_responses[0].subblock_power());
+  EXPECT_LT(0.0, fetch_subblock_responses[0].integration_bandwidth());
+  EXPECT_LT(0.0, fetch_subblock_responses[0].frequency());
+  EXPECT_LT(0.0, fetch_subblock_responses[1].subblock_power());
+  EXPECT_LT(0.0, fetch_subblock_responses[1].integration_bandwidth());
+  EXPECT_LT(0.0, fetch_subblock_responses[1].frequency());
+  EXPECT_GT(0.0, fetch_measurement_responses[0].lower_relative_power());
+  EXPECT_GT(0.0, (fetch_measurement_responses[0].upper_relative_power()));
```
Comment (mgumble) [inline source/tests/system/nirfmxnr_driver_api_tests.cpp:209]: couple extra parenthesis here and below

File: source/tests/system/nirfmxnr_driver_api_tests.cpp [hunk 4]
```diff
@@ -87,21 +123,622 @@ TEST_F(NiRFmxNRDriverApiTests, Init_Close_Succeeds)
...
+    compositePeakEVMSymbolIndex[i] = GET_ATTR_I32(session, carrier_string_response.selector_string_out(), NIRFMXNR_ATTRIBUTE_MODACC_RESULTS_COMPOSITE_PEAK_EVM_SYMBOL_INDEX);
+    compositePeakEVMSubcarrierIndex[i] = GET_ATTR_I32(session, carrier_string_response.selector_string_out(), NIRFMXNR_ATTRIBUTE_MODACC_RESULTS_COMPOSITE_PEAK_EVM_SUBCARRIER_INDEX);
+    componentCarrierFrequencyErrorMean[i] = GET_ATTR_F64(session, carrier_string_response.selector_string_out(), NIRFMXNR_ATTRIBUTE_MODACC_RESULTS_COMPONENT_CARRIER_FREQUENCY_ERROR_MEAN);
+    componentCarrierIQOriginOffsetMean[i] = GET_ATTR_F64(session, carrier_string_response.selector_string_out(), NIRFMXNR_ATTRIBUTE_MODACC_RESULTS_COMPONENT_CARRIER_IQ_ORIGIN_OFFSET_MEAN);
+    componentCarrierIQGainImbalanceMean[i] = GET_ATTR_F64(session, carrier_string_response.selector_string_out(), NIRFMXNR_ATTRIBUTE_MODACC_RESULTS_COMPONENT_CARRIER_IQ_GAIN_IMBALANCE_MEAN);
+    componentCarrierQuadratureErrorMean[i] = GET_ATTR_F64(session, carrier_string_response.selector_string_out(), NIRFMXNR_ATTRIBUTE_MODACC_RESULTS_COMPONENT_CARRIER_QUADRATURE_ERROR_MEAN);
+    switch (downlinkTestModel)
```
Comment (mgumble) [inline source/tests/system/nirfmxnr_driver_api_tests.cpp:390]: We should either parameterize around this enum or just delete the other cases

**Review by mgumble — APPROVED** [review]

**Pull request merged**